worlds/AutoWorld.py

```diff
@@ -357,32 +359,40 @@ def fill_hook(self,
         pass
 
     def post_fill(self) -> None:
-        """Optional Method that is called after regular fill. Can be used to do adjustments before output generation.
-        This happens before progression balancing, so the items may not be in their final locations yet."""
+        """
+        Optional Method that is called after regular fill. Can be used to do adjustments before output generation.
+        This happens before progression balancing, so the items may not be in their final locations yet.
+        """
 
     def generate_output(self, output_directory: str) -> None:
-        """This method gets called from a threadpool, do not use multiworld.random here.
-        If you need any last-second randomization, use self.random instead."""
+        """
+        This method gets called from a threadpool, do not use multiworld.random here.
+        If you need any last-second randomization, use self.random instead.
+        """
         pass
 
-    def fill_slot_data(self) -> Mapping[str, Any]:  # json of WebHostLib.models.Slot
-        """What is returned from this function will be in the `slot_data` field
+    def fill_slot_data(self) -> Dict[str, Any]:  # json of WebHostLib.models.Slot
```

Comment:
Not at all related to this PR but shouldn't this be modeled in AP proper somewhere since webhost already has strict dependency on other parts of AP but the converse should not be true (and technically it is not now, but it's also weakly typed)

Reply:
didn't realize i accidentally swapped that back to a `Dict` when merging conflicts. whoops. as for the typing being so weak we don't really have much of a choice due to how much is actually allowed https://github.com/ArchipelagoMW/Archipelago/pull/2876. worst case we do have a test to assert the data it passes is actually valid and the webhost will reject it if it's invalid as well.